### PR TITLE
Modified reduction example in Dask episode

### PR DESCRIPTION
**Pull request by PabRod**
Using `sum` both as a `map` and as a `reduction` was potentially confusing. Now we use different functions at each step (user-defined `f` for the mapping, `sum` for the reduction).

### CODE REVIEW

File: _episodes/04-dask-abstractions.md [hunk 1]
```diff
@@ -115,10 +115,10 @@ bag.filter(pred).compute()
 ### Reduction
 
 ~~~python
-bag.reduction(sum, sum).visualize()
+bag.reduction(f, sum).visualize()
```
Comment (jhidding) [inline _episodes/04-dask-abstractions.md:118]: Please read: https://docs.dask.org/en/latest/_modules/dask/bag/core.html#Bag.reduction
The bag is split in chunks, each chunk to be processed separately using function one, then these results are aggregated using the second function. If we compute a sum, then both functions should be sum.

**Comment by jhidding** [comment]
Why do you change a perfectly fine SVG into a PNG? I don't understand...

**Comment by PabRod** [comment]
I changed the content of the figure (see [here](https://github.com/escience-academy/lesson-parallel-python/blob/gh-pages/fig/dask-bag-reduction.png)) accordingly to the text.

It is a png just because that was the default exporting format 😅

File: _episodes/04-dask-abstractions.md (hunk 1 above)
Comment (PabRod) [reply _episodes/04-dask-abstractions.md:118]: Oh, I see. Then we can undo commit https://github.com/carpentries-incubator/lesson-parallel-python/pull/47/commits/bd851b549cee61f50f5eff0c720929ddc8da2759

**Comment by jhidding** [comment]
Ok; I still don't agree with the change though. In most simple cases the two functions used will be of the same nature: they're both reductions, one on the chunks, the other on the group of chunks: If the input is a list of strings, the chunking is changing the input in a list of list of strings that is then twice reduced: one time locally in each process, and one time globally to collect all the results.

File: _episodes/04-dask-abstractions.md (hunk 1 above)
Comment (PabRod) [reply _episodes/04-dask-abstractions.md]: Commit https://github.com/carpentries-incubator/lesson-parallel-python/pull/47/commits/c05baaf22ff747953c5c6cd30be51e4b708451d1 reverts the problematic commit

**Comment by PabRod** [comment]
Thanks!

**Pull request closed without merging**